File: scripts/hello_world.py
"""Script de teste do agrr — sem dependências externas."""

import sys
sys.path.insert(0, "sdk/python")

from agrr_sdk import AgrrScript


class HelloWorld(AgrrScript):
    name = "Hello World"
    description = "Exibe uma saudação — script de teste"
    group = "exemplos"
    version = "1.0.0"
    runtime = {"language": "python", "min_version": "3.8"}

    args = [
        {"name": "nome", "prompt": "Qual é o seu nome?"},
        {
            "name": "idioma",
            "prompt": "Idioma?",
            "options": ["pt", "en", "es"],
        },
    ]

    def run(self, creds: dict, args: dict) -> None:
        nome = args.get("nome") or "Mundo"
        idioma = args.get("idioma", "pt")

        saudacoes = {"pt": "Olá", "en": "Hello", "es": "Hola"}
        saudacao = saudacoes.get(idioma, "Olá")

        print(f"{saudacao}, {nome}!")
        print("Script de teste executado com sucesso. ✓")


if __name__ == "__main__":
    HelloWorld.main()

File: scripts/hello_world.js
'use strict';

const path = require('path');
// Resolve o SDK a partir da raiz do projeto para funcionar sem npm link.
const { createAgrrScript } = require(path.join(__dirname, '..', 'sdk', 'js', 'index.js'));

createAgrrScript({
  meta: {
    name: 'Hello World (JS)',
    description: 'Exibe uma saudação — script de teste Node.js',
    group: 'exemplos',
    version: '1.0.0',
    runtime: { language: 'node', min_version: '18' },
    args: [
      { name: 'nome', prompt: 'Qual é o seu nome?', type: 'text' },
      {
        name: 'idioma',
        prompt: 'Idioma da saudação?',
        type: 'select',
        options: ['pt', 'en', 'es'],
        default: 'pt',
      },
    ],
  },

  async run({ args }) {
    const nome = args['nome'] || 'Mundo';
    const idioma = args['idioma'] || 'pt';

    const saudacoes = { pt: 'Olá', en: 'Hello', es: 'Hola' };
    const saudacao = saudacoes[idioma] ?? 'Olá';

    console.log(`${saudacao}, ${nome}!`);
    console.log('Script JS executado com sucesso. ✓');
  },
});
